index.js

```diff
@@ -29,21 +29,45 @@ class LogStreamingPlugin {
 	}
 
 	logServerless() {
+		let logHandlerLambdaName;
+		const logHandlerFnName = _.get(this.serverless, "service.custom.logHandler.function", "loghandler");
 		_.forEach(this.serverless.service.getAllFunctions(), functionName => {
-			if (functionName === "loghandler") {
+			const functionLogicalId = `${_.upperFirst(functionName)}LambdaFunction`;
```

Comment:
I think you should use a Serverless function here to generate the name properly. Your implementation misses some character encodings.

Just use `this.serverless.service.provider.naming.XXXXX`. See https://github.com/serverless/serverless/blob/master/lib/plugins/aws/lib/naming.js for all available standardized naming methods.